js/import/rules.import.js

```diff
@@ -90,17 +123,29 @@ function buildTransformationRulesFromMapping(mapping) {
       return blockMap;
     }, {});
 
-  transformRules.blocks = [
-    ...transformRules.blocks,
-    ...Object.entries(blockMapping).map(([type, mappingList]) => {
-      const selectors = mappingList.map((m) => buildSelector(m, XPATH_BODY));
-      return {
-        type,
-        selectors,
-      };
-    })];
+  transformRules.blocks = Object.entries(blockMapping).map(([type, mappingList]) => {
+    const existingRules = transformRules.blocks.find((b) => b.type === type);
+    const selectors = mappingList.map((m) => buildSelector(m, rootXpath)).filter((s) => s);
+    const cells = buildBlockCellsFromMapping(mappingList);
+    return {
+      ...existingRules,
+      type,
+      selectors,
+      params: { cells },
+    };
+  });
```

Comment:
Builinding up the set of block rules for importing which is based on all the available mapping objects.